dmrpp_generator/main.py

```diff
@@ -61,29 +57,27 @@ def process(self):
         collection = self.config.get('collection')
         buckets = self.config.get('buckets')
         granules = self.input['granules']
-        append_output = {}
         for granule in granules:
-            granule_id = granule['granuleId']
+            dmrpp_files = []
             for file_ in granule['files']:
+                if not match(f"{self.processing_regex}$", file_['filename']):
+                    continue
                 output_file_path = self.dmrpp_generate(file_['filename'])
                 if output_file_path:
-                    s3_path = output_file_path.get('s3_path')
-                    file_local_path = output_file_path.get('file_local_path')
-                    append_output[granule_id] = append_output.get(granule_id, {'files': []})
-                    append_output[granule_id]['files'].append({
-                        "bucket": self.get_bucket(file_['filename'], collection.get('files', []),buckets)['name'],
-                        "filename": s3_path,
-                        "name": os.path.basename(file_local_path),
-                        "size": os.path.getsize(file_local_path),
+                    dmrpp_file = {
+                        "name": os.path.basename(output_file_path),
                         "path": self.config.get('fileStagingDir'),
-                        "url_path": self.config.get('fileStagingDir'),
+                        "url_path": file_['url_path'],
```

Comment:
Can we change this line to: `"url_path": file_.get('url_path', self.config.get('fileStagingDir'))`. Just in case where url_path is not defined in the payload and we don't don't really want to deal with that kind of error.